src/lib/kademlia/membership.ml

```diff
@@ -54,6 +54,43 @@ let get_project_root () =
   in
   go @@ realpath current_dir_name
 
+(* This snippet was taken from our fork of RPC Parallel.
+ * Would be nice to have a shared utility, but this is
+ * easiest for now. *)
+(* To get the currently running executable:
+  On Darwin:
+  Use _NSGetExecutablePath via Ctypes
+
+  On Linux:
+  Use /proc/PID/exe
+   - argv[0] might have been deleted (this is quite common with jenga)
+   - `cp /proc/PID/exe dst` works as expected while `cp /proc/self/exe dst` does not *)
+let get_coda_binary =
+  let our_binary_lazy =
+    lazy
+      (let open Async in
+      let open Deferred.Or_error.Let_syntax in
+      let%map os = Process.run ~prog:"uname" ~args:["-s"] () in
+      if os = "Darwin\n" then (
```

Comment:
you tested this on Mac right?

Reply:
@nholland94 	hook me up with instructions on how to test and I can do that for you

Reply:
Yes, it is a copy-paste of existing code we use that is just scoped to another library and hidden.